lte/gateway/c/oai/tasks/amf/Registration.cpp

```diff
@@ -652,23 +656,21 @@ static int registration_accept_t3550_handler(
       OAILOG_ERROR(
           LOG_AMF_APP,
           "Timer: timer has expired Sending Registration request again\n");
-      // amf_registration_request(registration_proc); /* TO DO for negative
+      amf_send_registration_accept(amf_ctx);
       // scenario */
+      OAILOG_FUNC_RETURN(LOG_NAS_AMF, RETURNok);
     } else {
       /* Abort the registration procedure */
-
       OAILOG_ERROR(
           LOG_AMF_APP,
           "Timer: Maximum retires done hence Abort the registration "
           "procedure\n");
-      return -1;
+      amf_remove_ue_context(ue_amf_context);
```

Comment:
what does the spec 24.501 suggest? is it enough if we remove UE context? or do we need to send any downlink NAS message in this case?